debian/update.sh

```diff
@@ -50,77 +50,51 @@ if [ ! -L "$INSTDIR/linux" ] ; then
 fi
 rm -rf "$INSTDIR/headers"
 
-# build CM1 kernel
-(cd linux; eval $make revpi-v6_defconfig)
-(cd linux; eval $make -j$NPROC zImage modules dtbs 2>&1)
-version=$(cat "$BUILDDIR/include/config/kernel.release")
-[ ! -d "extra" ] && mkdir "extra"
-echo "_ _ $version" > extra/uname_string
-copy_files
-
-# build CM1 piKernelMod
-if [ -d "$PIKERNELMODDIR" ] ; then
-	cd "$PIKERNELMODDIR"
-	make compiletime.h
-	cd -
-	(cd linux; eval $make M="$PIKERNELMODDIR" modules)
-fi
-# install CM1 kernel
-cp "$BUILDDIR/arch/arm/boot/zImage" "$INSTDIR/boot/kernel.img"
-
-# install CM1 modules
-rm -rf modules/*
-if [ -d "$PIKERNELMODDIR" ] ; then
-	(cd linux; eval $make -j$NPROC modules_install INSTALL_MOD_PATH="$INSTDIR/modules" M="$PIKERNELMODDIR")
-fi
-(cd linux; eval $make -j$NPROC modules_install INSTALL_MOD_PATH="$INSTDIR/modules")
-mv "$INSTDIR/modules/lib/modules"/* "$INSTDIR/modules"
-rm -r "$INSTDIR/modules/lib"
-rm "$INSTDIR/modules"/*/{build,source}
-
-# install CM1 dtbs
 [ -d "$INSTDIR/boot/overlays" ] || mkdir "$INSTDIR/boot/overlays"
 rm -f "$INSTDIR/boot"/*.dtb "$INSTDIR/boot/overlays"/*.dtbo
-(cd linux; eval $make -j$NPROC dtbs_install INSTALL_DTBS_PATH=/tmp/dtb.$$)
-mv /tmp/dtb.$$/*.dtb "$INSTDIR/boot"
-mv /tmp/dtb.$$/overlays/* "$INSTDIR/boot/overlays"
-rmdir /tmp/dtb.$$/overlays /tmp/dtb.$$
-
-# build CM3 kernel
-make+=7
-BUILDDIR+=7
-rm -rf "$BUILDDIR"
-mkdir "$BUILDDIR"
-(cd linux; eval $make revpi-v7_defconfig)
-(cd linux; eval $make -j$NPROC zImage modules dtbs 2>&1)
-version="$(cat "$BUILDDIR/include/config/kernel.release")"
-copy_files
-
-# build CM3 piKernelMod
-if [ -d "$PIKERNELMODDIR" ] ; then
-	cd "$PIKERNELMODDIR"
-	make compiletime.h
-	cd -
-	(cd linux; eval $make M="$PIKERNELMODDIR" modules)
-fi
 
-# install CM3 kernel
-cp "$BUILDDIR/arch/arm/boot/zImage" "$INSTDIR/boot/kernel7.img"
-
-# install CM3 modules
-if [ -d "$PIKERNELMODDIR" ] ; then
-	(cd linux; eval $make -j$NPROC modules_install INSTALL_MOD_PATH="$INSTDIR/modules" M="$PIKERNELMODDIR")
-fi
-(cd linux; eval $make -j$NPROC modules_install INSTALL_MOD_PATH="$INSTDIR/modules")
-mv "$INSTDIR/modules/lib/modules"/* "$INSTDIR/modules"
-rm -r "$INSTDIR/modules/lib"
-rm "$INSTDIR/modules"/*/{build,source}
-
-# install CM3 dtbs
-(cd linux; eval $make -j$NPROC dtbs_install INSTALL_DTBS_PATH=/tmp/dtb.$$)
-mv /tmp/dtb.$$/*.dtb "$INSTDIR/boot"
-mv /tmp/dtb.$$/overlays/* "$INSTDIR/boot/overlays"
-rmdir /tmp/dtb.$$/overlays /tmp/dtb.$$
+# CM1=6 CM3=7 CM4=7l
+kernel_versions="6 7 7l"
+
+for kernel_version in $kernel_versions; do
+    rm -rf "$BUILDDIR"
```

Comment:
Until this commit, we have separate build directories for the CM1 and CM3. It is advantageous to retain that in order to support incremental builds. (I frequently run incremental builds manually after inserting debug code or to test fixes.) Because the build directory is deleted at the beginning of the loop in your commit, incremental builds are only possible for the v7l variant.

So please append `$kernel_version` to `$BUILDDIR` at the beginning of the loop and strip it at the end of the loop. (It probably doesn't matter whether the CM1 directory is called `kbuild` or `kbuild6`.)

Reply:
I moved the build dir to a local variable for each individual build based on the default in `BUILDDIR` (version `6` is treated as default value and the `6` is removed from path)